lib/engine/step/dividend.rb

```diff
@@ -88,13 +89,29 @@ def payout(entity, revenue)
         { company: 0, per_share: payout_per_share(entity, revenue)[0] }
       end
 
+      def dividends_for_entity(entity, holder, per_share)
+        return 0 if (shares = holder.num_shares_of(entity)).zero?
```

Comment:
can't this just be holder.num_shares_of(entity) * per_share, why do you need to check for zero

Reply:
Sure, I was just changing the existing code as little as possible, but it used to have more of a reason to early-out.